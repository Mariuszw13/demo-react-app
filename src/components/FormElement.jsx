import React from 'react';
import Form from "react-bootstrap/Form";
import Row from "react-bootstrap/Row";
import Col from "react-bootstrap/Col";

const FormElement = ({label, error, controlProps}) => (
    <Form.Group as={Row}>
        <Form.Label column sm="2">
            {label}
        </Form.Label>
        <Col sm="10">
            <Form.Control placeholder={label} isValid={String(controlProps.value).length && !error} {...controlProps}/>
        </Col>
    </Form.Group>
);

export default FormElement;