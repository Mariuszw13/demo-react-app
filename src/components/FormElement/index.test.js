import React from "react";
import { render } from "@testing-library/react";
import FormElement from "./index";

describe("<FormElement />", () => {
    const label = "test label";
    const error = null;
    const controlProps = {};

    let getByText, container;
    beforeEach(() => {
        const utils = render(<FormElement label={label} error={error} controlProps={controlProps} />);
        getByText = utils.getByText;
        container = utils.container;
    });

    it("should render properly and match snapshot", () => {
        expect(container.firstChild).toMatchSnapshot();
    });

    it("should render label", () => {
        expect(getByText(label)).toBeInTheDocument();
    });
});
