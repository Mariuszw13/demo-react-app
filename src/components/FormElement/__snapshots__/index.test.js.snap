// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormElement /> should render properly and match snapshot 1`] = `
<div
  class="form-group row"
>
  <label
    class="form-label col-form-label col-sm-2"
  >
    test label
  </label>
  <div
    class="col-sm-10"
  >
    <input
      class="form-control is-valid"
      placeholder="test label"
    />
  </div>
</div>
`;
